### DISCUSSION TITLE
PLEASE HELP! AIR 50 SDK INVALID INPUT PROBLEM

### DISCUSSION THREAD

**Discussion by hantoro-05** [airsdk/Adobe-Runtime-Support#2302] [General]
Hello! I encountered a publishing problem on my Adobe Animate. I'm currently making an Android game and every time I try to publish it it is says "INVALID INPUT" and that the adl can't be check for malicious malware and that the software needs an update :( I don't know how to solve this. I tried redownloading the AIRSDK 50 in case it would work but it remains the same.

Is this related to updating my macbook ios to MacOS Big Sur? Whether it is or not someone please help :( what im currently working on is an assignment so it really urgent and I don't have any clue how to solve this :( 

List of things I have tried doing:

Redownloading AIRSDK
Allowing the files of the AIR SDK in my security and privacy

Thank you :(

**Comment by ajwfrost** [comment]
Can you open a terminal window and go into the folder where the AIR SDK is, and then type:
`xattr -d -r com.apple.quarantine AIRSDK_MacOS` (or whatever the folder is called..)

Mostly like because it's downloaded from the internet, the Apple security will block some of the binaries..

If you do `ls -la` you may see an `@` symbol against files if they have these extended attributes, or run `xattr` on its own to see what files have them...

thanks

**Reply by hantoro-05** [reply]
Hello! Thank you for the reply...not sure though what a terminal window is exactly :( Sorry...may you please elaborate on how to find it or what I should be looking for? Thank you

**Reply by ajwfrost** [reply]
If you use the Spotlight thing (magnifying glass on top-right of screen, or Command-Space on the keyboard) and type "termin" you'll see it..
![image](https://user-images.githubusercontent.com/6400804/202714988-bfcb6ee1-d855-4dbf-a9a0-8ed95eb92236.png)

I'm just downloading the latest AIR SDK to that machine, will show you how it looks and how to un-quarantine it once the download has finished..

**Reply by hantoro-05** [reply]
Oh okay I did open that a while ago and tried what you said. Although I'm not sure if what I did was right as it said that there's no such file existing my system (although the file of the AIRSDK is on my desktop) here is the photo: 
<img width="278" alt="Screen Shot 2022-11-18 at 9 26 48 PM" src="https://user-images.githubusercontent.com/118537174/202716096-aae99579-2766-4f55-a8dc-049fd0e6230d.png">

**Reply by hantoro-05** [reply]
Thank you again for trying to help me! I really do hope I can solve this issue :<

**Reply by ajwfrost** [reply]
Okay so if it's on the desktop then it may need adjusting a little. Below is my screenshot, the 'normal' AIR SDK in the Downloads folder.
![image](https://user-images.githubusercontent.com/6400804/202716534-640b66ae-5205-4238-9e0c-1d2dff54170d.png)
(see the `@` symbol against each item.. they will go if the quarantine or extended attributes are removed)

If you've got it in the Desktop then you could try
```
cd ~/Desktop
ls AIRSDK*
```
and it should then show what the actual filename is, so then the xattr command should be used with that folder name..

**Reply by hantoro-05** [reply]
<img width="360" alt="Screen Shot 2022-11-18 at 9 40 39 PM" src="https://user-images.githubusercontent.com/118537174/202718167-610061a9-fa6f-40ad-88c2-33a29e5e4295.png">

Is this right so far??  If so, may I know which one is considered the actual filename?  Is it still AIRSDK_Flex_MacOS?

**Reply by ajwfrost** [reply]
Also -> this was one of the issues that was behind the idea of the AIR SDK Manager, you could try using that to download the SDK. See #2291

**Reply by ajwfrost** [reply]
So yes, looks like you've still got the zip but also the expanded folder, so:
```
xattr -r -d com.apple.quarantine AIRSDK_Flex_MacOS
```

**Reply by hantoro-05** [reply]
Oh okay! I see the @ symbol now! So how do I remove the aforementioned extended attributes?
<img width="693" alt="Screen Shot 2022-11-18 at 9 51 08 PM" src="https://user-images.githubusercontent.com/118537174/202720312-0190b19e-3069-4745-a7b7-a0846c224e3d.png">

**Reply by ajwfrost** [reply]
I would have thought that your first command there should have been enough! Can you run:
```
xattr AIRSDK_Flex_MacOS
```
to see what the remaining attribute is?

Slightly concerning that the folder is not even readable by anyone else, but as long as it's just you using the machine, it should be okay

**Reply by hantoro-05** [reply]
<img width="449" alt="Screen Shot 2022-11-18 at 9 55 56 PM" src="https://user-images.githubusercontent.com/118537174/202721393-f6b36fec-a0c7-4211-babb-61b07686bc9b.png">

This is what came out: "com.apple.macl"

**Reply by ajwfrost** [reply]
Okay, I think that `macl` can be ignored .. it's removed the quarantine flag which is good. So can you check the functionality again to see if it works this time..?

thanks

**Reply by hantoro-05** [reply]
Will do! Thank you very much, Mr. Frost. I'll update you in a bit.

**Reply by hantoro-05** [reply]
Hello!!! I think there's been an improvement because the "Invalid Input" issue is gone!! :)) However, a new error popped up.
<img width="244" alt="Screen Shot 2022-11-18 at 10 06 34 PM" src="https://user-images.githubusercontent.com/118537174/202723349-efd436aa-495b-4f0c-bab9-1f1648a9d2c4.png">

**Reply by hantoro-05** [reply]
hmmm i spotted the adt.jar in the expanded file of the AIR SDK but Im not sure how to solve it...

Is it possible that you can still help me with this one? @ajwfrost It would mean a lot to me, thank you :(

**Reply by ajwfrost** [reply]
Sorry for the delay. I'm wondering about permissions here. Firstly though, have you copied this folder over to under the Animate location? I don't think you need to do that, you can have the SDK wherever you want and just get Animate to find it with the "Manage AIR SDK" option under the Help menu.

But if you can open up a terminal window and do:
```
cd "/Applications/Adobe Animate 2023/AIR50.0/lib"
ls -la
```
then it should get you to that folder and list everything with all details. If it can't find the folder, then there's a fundamental problem that Animate is just looking in the wrong place! so maybe that "Manager AIR SDK" can be used.

Otherwise I suspect there may be an entry for adt.jar but without the appropriate permissions. Maybe try
```
chmod a+rx adt.jar
```
to give everyone read and execute access for it? You may need to use `sudo chmod...` as it's in the /Applications folder, if you type that (super-user do something) then you'll be asked for your login password..

thanks

**Reply by hantoro-05** [reply]
Hmmm...well, I didn't copy the folder. I only use the help and manage AIR SDK option for it to be available on my Adobe Animate Software.

And perhaps it is looking in the wrong place because this is what it says:
<img width="569" alt="Screen Shot 2022-11-18 at 11 19 29 PM" src="https://user-images.githubusercontent.com/118537174/202738662-618b5a4d-b945-4d47-a328-e233237a631b.png">

I just checked 'Manage Air SDK' and it says there that the version there is 50.0.0.0 (this is the previous version i had downloaded but it's gone now) This explains why the output message on adobe animate was a warning about how it was not there anymore. Should I try to replace it with the latest version 50.0.1.2, first? 

thank you

**Reply by hantoro-05** [reply]
When I command return to check the preview of my game, it works! Which is good because it didn't work prior... hmmm but now it says invalid input again :<

**Comment by hantoro-05** [comment]
<img width="602" alt="Screen Shot 2022-11-18 at 11 29 38 PM" src="https://user-images.githubusercontent.com/118537174/202741294-ae86a5c1-9150-4987-8fe0-ae0cfd4eee0a.png">

The first photo is the update on the "invalid input situation"

<img width="385" alt="Screen Shot 2022-11-18 at 11 30 46 PM" src="https://user-images.githubusercontent.com/118537174/202741301-f6d77e83-648d-4c28-9f47-f8131015d5a3.png">

The second photo is what happened when I tried the sudo chmod...

**Reply by ajwfrost** [reply]
You need to change into the folder/directory where those files are .. so in that first screenshot you can see all the folders within your home folder (`~`), the AIRSDK_Flex_MacOS folder was under `Desktop` I believe so change directory using `cd Desktop`

Then the adt.jar file is within the SDK's "lib" folder so you need to do `cd lib`

Then the chmod command should work (without sudo, in this case). The sudo thing btw, I was lazily not writing the full command so put `...` after the start of it. So if you do find it tells you that you don't have permissions for changing the permissions, you'd need to do `sudo chmod a+rx adt.jar` i.e. just prefix whatever command you'd been trying with `sudo `...

thanks

**Reply by hantoro-05** [reply]
Is the computer perhaps confused because one of the two versions of AIRSDK is in the trash?

**Reply by hantoro-05** [reply]
> You need to change into the folder/directory where those files are .. so in that first screenshot you can see all the folders within your home folder (`~`), the AIRSDK_Flex_MacOS folder was under `Desktop` I believe so change directory using `cd Desktop`
> 
> Then the adt.jar file is within the SDK's "lib" folder so you need to do `cd lib`
> 
> Then the chmod command should work (without sudo, in this case). The sudo thing btw, I was lazily not writing the full command so put `...` after the start of it. So if you do find it tells you that you don't have permissions for changing the permissions, you'd need to do `sudo chmod a+rx adt.jar` i.e. just prefix whatever command you'd been trying with `sudo `...
> 
> thanks

Oh i see! okay then wait let me try

**Reply by hantoro-05** [reply]
Oh no Im so sorry...I'm quite lost :<

Is it right that I just type in cd Desktop then proceed to typing cd lib in the terminal? 

Also, I'm not quite sure what the function or purpose of sudo and chmod is...sorry if I sound so silly. I'm really just new and confused by all of this :(

**Reply by hantoro-05** [reply]
<img width="528" alt="Screen Shot 2022-11-18 at 11 51 37 PM" src="https://user-images.githubusercontent.com/118537174/202745975-500682fb-16f3-4cc2-8f7e-74e9409540f2.png">

I don't think I'm doing it right...

**Reply by ajwfrost** [reply]
> I don't think I'm doing it right...

:-) so, yes the folder hierarchy I think would be `~/Desktop/AIRSDK_Flex_MacOS/lib/` so you'd need to do them in order:
```
cd ~
cd Desktop
cd AIRSDK_Flex_MacOS
cd lib
```
It shows you, after your computer name and before your username, what folder you're in.

> Is the computer perhaps confused because one of the two versions of AIRSDK is in the trash?

Which one is in the trash?! If you've trashed the initial AIR 50.0 release that was showing in Animate as 50.0.0.0, then you'll need to go to the Animate Help -> Manage AIR SDKs, delete that one if it's still there, and then add a new one and browse to your Desktop/AIRSDK_Flex_MacOS folder...

**Reply by hantoro-05** [reply]
<img width="443" alt="Screen Shot 2022-11-19 at 12 18 54 AM" src="https://user-images.githubusercontent.com/118537174/202751710-3ed5febc-a079-42c8-8f49-e06bf8c80a09.png">

 Oh okay so it says here that it's in the library under the noemiagustin folder
:OOO

**Reply by hantoro-05** [reply]
> > I don't think I'm doing it right...
> 
> :-) so, yes the folder hierarchy I think would be `~/Desktop/AIRSDK_Flex_MacOS/lib/` so you'd need to do them in order:
> 
> ```
> cd ~
> cd Desktop
> cd AIRSDK_Flex_MacOS
> cd lib
> ```
> 
> It shows you, after your computer name and before your username, what folder you're in.
> 
> > Is the computer perhaps confused because one of the two versions of AIRSDK is in the trash?
> 
> Which one is in the trash?! If you've trashed the initial AIR 50.0 release that was showing in Animate as 50.0.1, then you'll need to go to the Animate Help -> Manage AIR SDKs, delete that one if it's still there, and then add a new one and browse to your Desktop/AIRSDK_Flex_MacOS folder...

Also, the earlier version 50.0.0 is the one i replaced and threw and I updated my Adobe Air SDK settings to the latest one which is 50.0.1.2

**Reply by hantoro-05** [reply]
By replacing the AIRSDK and updating it on adobe animate, there was an improvement since Im now able to use the test movie feature but when I publish it, it's back with the previous error that says "input invalid"

**Reply by hantoro-05** [reply]
<img width="559" alt="Screen Shot 2022-11-19 at 12 20 14 AM" src="https://user-images.githubusercontent.com/118537174/202752035-8745e2a2-d318-4d3c-bc4f-a444701ae272.png">

Here's what happened when I entered the chmod command

**Reply by hantoro-05** [reply]
not sure what to do next and im not sure if the chmod result is righttt x((( oh my goodness im so sorry that this is taking so long. Thank you for being patient

**Reply by ajwfrost** [reply]
Okay -> so it sounds like a lot of things are set up properly now, if you are able to use the "test movie" feature, then it's got the debug launcher and the runtime all set up properly. If it's telling you that it's version 50.0.1.2 then it's actually okay regarding the permissions on the adt.jar file, since that's called by Animate to get the version.

So ... can you share the screenshot of the "Input Invalid" thing? I'm wondering what that might be. I'm assuming that you're just testing a simple/initial FLA file so there's nothing particularly complex that you've added?

thanks

**Reply by hantoro-05** [reply]
<img width="251" alt="Screen Shot 2022-11-19 at 12 46 01 AM" src="https://user-images.githubusercontent.com/118537174/202757704-900e47cb-a287-4ad2-ab2b-8fd03ef5a9a2.png">

Yes I just put simple motion tweens and one basic code snippet that is "Stop and Go to Frame"

**Reply by ajwfrost** [reply]
Hmmm. Well the warning and the note are things that get spat out by the ADT java app when it's creating an Android package. "Invalid input" sounds more ominous though. Presumably it's not actually generating any APK file after this?

The best way to debug this next bit is to try to work out what Animate is calling in ADT and to capture the ADT logs. So if you use Finder to find the AIR SDK folder, then the "lib" folder in that, there should be a file "adt.cfg" which is a text file: hopefully you can open it with the built-in text editor? There's a line:
```
#DebugOut=true
```
where you need to remove the `#` symbol at the start, then save it.

Then do the Animate publish thing again (no need to restart it or anything..)

Then in your Finder, go to your home folder (up one from Documents or Desktop) and there should be a file "adt.log". If you can copy/paste that into this thread it should tell us what arguments are being passed by Animate, and which of them is causing the invalid input...

thanks

**Reply by hantoro-05** [reply]
*** ADT Logging at 2022/11/19 01:44:12 ***
Called by: /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/adt.jar
ADT 50.0.1.2 called with: -version
*** ADT Logging at 2022/11/19 01:47:23 ***
Called by: /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/adt.jar
ADT 50.0.1.2 called with: -package -target apk-captive-runtime -arch armv7 -storetype PKCS12 -keystore /Users/noemiagustin/Desktop/School 4th year/Game Dev Music App/Agustin_Hansel Genein_Music App.p12 -storepass xxxxxx /Users/noemiagustin/Desktop/School 4th year/Game Dev Music App/Agustin_Hansel Genein_Music App.apk /Users/noemiagustin/Desktop/School 4th year/Game Dev Music App/Agustin_Hansel Genein_Music App-app.xml -C /Users/noemiagustin/Desktop/School 4th year/Game Dev Music App/ Agustin_Hansel Genein_Music App.swf Icon.png
Not requested a legacy APK build, but could not find Javac and/or Android SDK hence falling back to legacy build
AAPT command line: /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/bin/aapt package -z -u -J /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50 -M /var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50/AndroidManifest.xml -S /var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50/app_entry_res -S /var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50/captive_runtime_res --auto-add-overlay -I /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/lib/resources/android-res.jar --no-version-vectors --min-sdk-version 14 --target-sdk-version 31 --version-code 1000000 --version-name 1.0.0 -F /var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50/resources.apk
Generating class file:  /Applications/Adobe Animate 2023/Adobe Animate 2023.app/Contents/jre/jdk-11.0.15.1.jdk/Contents/Home/bin/javac -encoding utf-8 -J-Xmx1024m -source 1.8 -target 1.8 -Xlint /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50/R.java -d /var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/5f70e038-530b-4d18-bc0b-a6eb48dc05d0
DX tool command line: java -Xmx1024m -cp /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/bin/d8.jar com.android.tools.r8.D8 --release --main-dex-list /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/lib/multidex-1.0.2/addToPrimaryDex.keep --output /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/57a20b31-b42a-47c8-985d-1b40cea64e48 /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/lib/multidex-1.0.2/classes.jar /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/57a20b31-b42a-47c8-985d-1b40cea64e48/inputJAR.jar /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/19e6601d-98bf-4e6b-b97a-b31066520a50/resources.jar /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/lib/runtimeClasses.jar
Baksmali tool command line: java -jar /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/lib/baksmali.jar d -o /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/57a20b31-b42a-47c8-985d-1b40cea64e48/out /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/57a20b31-b42a-47c8-985d-1b40cea64e48/classes.dex
Smali tool command line: java -jar /Users/noemiagustin/Desktop/AIRSDK_Flex_MacOS/lib/android/lib/smali.jar a -o /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/57a20b31-b42a-47c8-985d-1b40cea64e48/out/classes.dex /private/var/folders/29/srq4fn7n30b7ct2_zv8vgz5h0000gn/T/57a20b31-b42a-47c8-985d-1b40cea64e48/out
Version in APK = 50.0.1.2, version of ADT = 50.0.1.2
..java.base/java.lang.Thread.getStackTrace(Thread.java:1606)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.InvalidInputException.<init>(InvalidInputException.java)
..com.adobe.air.ADTOutputStream.checkPath(ADTOutputStream.java)
..com.adobe.air.ADTOutputStream.addFile(ADTOutputStream.java)
..com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
Invalid input: The file with the path Icon.png cannot be packaged unless the descriptor field application.icon.image57x57 is set to Icon.png

Here's what it says :O So i may be wrong but it has something to do with the icon i chose for my published app?

**Reply by hantoro-05** [reply]
Mr. Frost, you got it!!! The icon for the game was the one causing the trouble this whole time!!! Is there a way to make it valid somehow though?

**Reply by hantoro-05** [reply]
A success!!!! oh my goodness thank you so much!!!! I was already on the verge of tears an hour ago. This means so much! The icon works now as well I just had to rename it and save a new one!

**Reply by hantoro-05** [reply]
Thank you very much!!!! :(( Thank you for solving the problem and also for dedicating the past couple of hours to guide me through it. You have no idea how much you've helped me!

**Reply by ajwfrost** [reply]
That's great - I was just going to look up exactly how that error message came about, you may need to play a little to work out how to set the right icons for the right devices etc. But good to hear it's working :-)

**Reply by hantoro-05** [reply]
Thank you very much! :^)) I look forward to working on my game again :) <3 <3 <3
